src/dmd/dstruct.d

```diff
@@ -606,39 +606,36 @@ extern (C++) class StructDeclaration : AggregateDeclaration
         return index < numArgTypes() ? (*argTypes.arguments)[index].type : null;
     }
 
-    final bool hasNonDisabledCtor()
+    /*
+       A constructor is regular if it is not a copy constructor.
+       Additionally, if the `checkDisabled` parameter is set,
+       the function will return true if there is a regular constructor
+       that is not disabled.
+     */
+    final bool hasRegularCtor(bool checkDisabled = false)
     {
-        static extern (C++) class HasNonDisabledCtorVisitor : Visitor
-        {
-            bool result;
-
-            this() {}
-
-            alias visit = Visitor.visit;
-
-            override void visit(CtorDeclaration cd)
-            {
-                if (!(cd.storage_class & STC.disable))
-                    result = true;
-            }
+        if (!ctor)
+            return false;
 
-            override void visit(TemplateDeclaration td)
+        bool result;
+        overloadApply(ctor, (Dsymbol s)
+        {
+            if (s.isTemplateDeclaration())
```

Comment:
Can a template constructor be disabled ?

Reply:
Yes it can, however, I saw that the previous implementation was not checking for this so I simply went it that.